### PR TITLE
Prevent importing styles with no section. Fixes #687

### LINKED ISSUE #687: Firefox: all the styles disappeared but still work
* **Browser**: Firefox 65.0.1 (64-bit)
* **Operating System**: Ubuntu 18.04.2 LTS
* **Stylus Version**: 1.5.3
* **Screenshots**: 
Stylus Dashboard
Stylus Popup Window

When I tried to import the following JSON, all the styles disappeared from both the dashboard and the popup window. Nonetheless, the add-on works.
```[json]
[
    {
        "name": "New Style"
    }
]
```

As soon as I changed the JSON to
```[json]
[
    {
        "name": "New Style",
        "sections": []
    }
]
```
, all the styles came back.

The problem repeates every time.

### PR DESCRIPTION
**Pull request by Mottie**
And prevent issues with existing styles with no section

### CODE REVIEW

**Comment by eight04** [comment]
It looks like this PR not just prevent no-section styles from being imported, but also accept no-section styles in the database. Is this intended?

**Comment by Mottie** [comment]
It doesn't accept no-section styles; but it doesn't throw an error if a style doesn't include a section - in case a style without a section was already imported.

**Comment by eight04** [comment]
I think we don't have to prevent it from throwing since it is not a valid use case?

**Comment by Mottie** [comment]
You're right, it's not a valid use case, but I don't think it would hurts anything to ensure that an empty section won't cause a JS error.

**Comment by eight04** [comment]
It hurts if some part of the code base don't expect the database to return no-section styles. As a developer, I will expect the system to throw an error when the database is broken (i.e. containing a no-section style).

**Comment by Mottie** [comment]
Ok, updated.

File: manage/import-export.js [hunk 1]
```diff
@@ -148,7 +148,9 @@ function importFromString(jsonString) {
         !item ||
         !item.name ||
         !item.name.trim() ||
-        (item.sections && !Array.isArray(item.sections))) {
+        !item.sections ||
+        !Array.isArray(item.sections)
```
Comment (eight04) [inline manage/import-export.js]: Do we want to check `item.sections.length`?

Comment (Mottie) [reply manage/import-export.js]: Probably. But then if someone creates a style as follows:

```js
[
    {
        "name": "New Style",
        "sections": [{}]
    }
]
```

It'll still break the editor. Should we add an additional check that `item.sections` contains an object and then check for `item.sections.code`?

**Comment by eight04** [comment]
I found that we have a `styleJSONSeemsValid` function in `update.js`:
https://github.com/openstyles/stylus/blob/b48817fe01d4079b5eeec0e817e3ed6f240a16f9/background/update.js#L235

We should probably pull it out as a standalone module and use it in import-export.js.

**Comment by eight04** [comment]
We can put the function in section-util.js.

**Comment by eight04** [comment]
We can delete `styleJSONSeemsValid` function inside update.js since it is now a global function.

**Pull request merged**